Sources/CNIOLinux/liburing_shims.c

```diff
@@ -0,0 +1,125 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2021 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+// Support functions for liburing
+
+// Check if this is needed, copied from shim.c to avoid possible problems due to:
+// Xcode's Archive builds with Xcode's Package support struggle with empty .c files
+// (https://bugs.swift.org/browse/SR-12939).
+void CNIOLinux_i_do_nothing_just_working_around_a_darwin_toolchain_bug2(void) {}
+
+#ifdef __linux__
+
+#ifdef SWIFTNIO_LIBURING
+
+#define _GNU_SOURCE
+#include <CNIOLinux.h>
+#include <pthread.h>
+#include <unistd.h>
+#include <sys/syscall.h>
+#include <stdlib.h>
+#include <assert.h>
+#include <signal.h>
+#include <signal.h>
+#include <sys/poll.h>
+
+// Definitons from syscall.c in liburing, we'll just do a syscall that will fail
```

Comment:
do we need this in this PR? If people don't have io_uring, then they shouldn't enable `SWIFTNIO_LIBURING`?

Reply:
You mean the check for the syscall etc? I guess not strictly needed, just defensive - can remove it. I guess at this point, anyone trying it out will be a brave soul anyway and know what they are doing...

Reply:
@Lukasa WDYT? I'd personally think start with the least amount of code and add in once we have people screaming rather than being overly defensive.

Reply:
(but I'm cool with syscall testing too)

Reply:
Ok, removing it.